File: Chapter1/S1.4/Python/exercise06.py
import pandas as pd
import matplotlib.pyplot as plt

num_kids = [3, 4, 3, 4, 5, 1, 6, 3, 4, 5, 3, 4, 3, 3, 4, 3, 5, 5, 5, 5, 6, \
            11, 10, 2, 1, 2, 3, 1, 5, 2]
                
# Build table with the frequency, percentage, and cumulative percentage
num_kids_freq = pd.Series(num_kids).value_counts()
num_kids_freq_norm = pd.Series(num_kids).value_counts(normalize=True)

df = pd.DataFrame()
df["Kids per Family Frequency"] = num_kids_freq
df["Relative"] = num_kids_freq_norm
df = df.sort_index()
df["Relative Cumulative"] = df["Relative"].cumsum()

print("Number of kids per family distribution frequency")
print(df)

plt.figure(num='Section 1.4 exercise 6')
plt.title("Distribution of number of kids per family")
plt.xticks(num_kids_freq.index)
plt.bar(num_kids_freq.index, num_kids_freq)
plt.show()

File: Chapter1/S1.4/R/exercise06.r
num_kids <- c(3, 4, 3, 4, 5, 1, 6, 3, 4, 5, 3, 4, 3, 3, 4, 3, 5, 5, 5, 5, 6, 11, 10, 2, 1, 2, 3, 1, 5, 2)

# Build table with the frequency, percentage, and cumulative percentage
num_kids_freq = table(num_kids)
num_kids_freq_norm = num_kids_freq / length(num_kids)
num_kids_freq_norm_cum = cumsum(num_kids_freq_norm)

df = cbind(num_kids_freq, num_kids_freq_norm, num_kids_freq_norm_cum)
colnames(df) <- c("Kids per Family Frequency", "Relative", "Relative Cumulative")

print("Number of kids per family distribution frequency")
print(df)

# The default barplot function suppress the x values when no totals are present.
# In this example there are no families with 7, 8, or 9 kids. I want these to be shown
# as empty spaces on the chart.

# Get the maximum and minimum number of kids per family, to build a range with them later
minX <- min(as.numeric(rownames(df)))
maxX <- max(as.numeric(rownames(df)))
# Build a new table using the minX and maxX as a range for the categories
num_kids_complete = table(factor(num_kids, levels=minX:maxX))

barplot(num_kids_complete, main="Distribution of number of kids per family")